toolbox/library/src/OSQP.cpp

```diff
@@ -350,11 +366,21 @@ bool wbt::block::OSQP::initialize(BlockInformation* blockInfo)
     ok = ok && m_parameters.getParameter("UseUb", pImpl->useUb);
     ok = ok && m_parameters.getParameter("ComputeObjVal", pImpl->computeObjVal);
     ok = ok && m_parameters.getParameter("StopWhenFails", pImpl->stopWhenFails);
+    ok = ok && m_parameters.getParameter("AdaptiveRho", pImpl->adaptiveRho);
+    ok = ok && m_parameters.getParameter("Polish", pImpl->polish);
+    ok = ok && m_parameters.getParameter("MaxIterations", maxIter);
 
     if (!ok) {
         bfError << "Failed to get parameters after their parsing.";
         return false;
     }
+    
+    // Check the maximum number of Iterations
+    if (maxIter > 0){
+       pImpl->maxIterations = maxIter;
+    } else {
+       bfError << "The maximum number of iterations is not valid.";
```

Comment:
Probably it is helpful to either `return false` or explain that we are going to use the maxIter value of 1 ?

Reply:
I think using the default value of 1 is a nice idea. Please see a3fa2f82b01ac457184622ebbd5fb40fac3baa1a

Reply:
I agree with @traversaro that is better to make the block fail rather than applying something the user has not explicitly configured (example: I set -2 and magically the block sets 1. I'd prefer a failing simulation that tell me that I have to set an integeter > 0). The semantics of `bfError` is the scary red warning in the simulink GUI, and the model should fail in this case.

Reply:
Furthemore, as replied in https://github.com/robotology/wb-toolbox/pull/234#pullrequestreview-1145947757, choosing a default value for the number of iterations that differs from the previous state (that was undefined, does OSQP has a documented number of iterations when not set?) would make this PR not backward compatible.

I suggest the following solutions:

- If OSQP has a documented number of iterations that is selected when the user does not specify it (-> what we were doing before this PR), use this number as default
- Othetwise, initialize in the mask an invalid number of iterations (e.g. -1) and do not call the `setMaxIteration` if it's less than 1. In this case, we can notify the user in the GUI with a message with low verbosity (lower than warning)

Reply:
Actually, I think we don't need to check the validity of the maxIterations in the C++ library, because in the Simulink GUI, 
- I defined an initial value (which is equal to 100 but I think we need to set it to 4000 for backward compatibility),
- I defined some constraints for the maximum iteration parameter, so in reality, the user can only choose a positive real number otherwise, the Simulink GUI generates an error.

So, I think we could remove this check from the C++ side.